.github/workflows/e2e.yml

```diff
@@ -0,0 +1,86 @@
+name: E2E Integration Tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  test:
+    name: E2E tests
+    runs-on: ubuntu-latest
+    env:
+      GAS_OUT_DIR: gas_reports
+      ENABLE_MAX_COLLECTION: true
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Install latest nightly toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          target: wasm32-unknown-unknown
+          override: true
+
+      - name: Rust Dependencies Cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+            artifacts/
+          key: ${{ runner.os }}-cargo-with-artifacts-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Deploy local Stargaze node
+        run: make deploy-local
+      
+      - name: Optimize Contracts
+        run: make optimize
+
+      - name: Run Integration Tests
+        run: make e2etest
+
+      - name: Combine Test Gas Reports
+        run: cd e2e/ && jq -rs 'reduce .[] as $item ({}; . * $item)' gas_reports/*.json > gas_report.json
+
+      - name: Raw Gas Report
+        run: cat e2e/gas_report.json
+
+      - name: Set GIT_BRANCH
+        run: echo "GIT_BRANCH=$(echo ${{ github.ref }} | sed 's|/|-|g')" >> $GITHUB_ENV
+
+      - name: Upload Gas Report
+        if: ${{ github.ref == 'refs/heads/main' }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: launchpad-gas-report-${{ env.GIT_BRANCH }}
+          path: e2e/gas_report.json
+          retention-days: 90
+
+      - name: Download main gas report
+        id: download_gas
+        # Because the max retention period of github artifacts is 90 days
+        # there's a possibility the main's report no longer exists
+        continue-on-error: true
+        if: ${{ github.ref != 'refs/heads/main' }}
+        # NOTE: We can't use github's `actions/download-artifact` because it doesnt support
+        # downloading an artifact cross workflows yet
+        # https://github.com/actions/download-artifact/issues/3
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          branch: main
+          workflow: e2e.yml
+          name: launchpad-gas-report-refs-heads-main
+
+      - name: Post gas diff to PR
+        if: ${{ github.ref != 'refs/heads/main' && steps.download_gas.outputs.found_artifact == 'true' }}
+        uses: de-husk/cosm-orc-gas-diff-action@v0.6.2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          current_json: e2e/gas_report.json
+          old_json: gas_report.json
```

Comment:
Once this is merged (and there is a gas usage report for main), this step will post the gas diff between this PR and the latest report uploaded from main branch.

Reply:
The github action will always post the Gas Usage as a [job summary](https://github.com/DA0-DA0/dao-contracts/actions/runs/3670836555/attempts/1#summary-10023289693), and if the user making the PR has write permissions to the `public-awesome/launchpad` repo, the action will post this as a PR comment too.